File: tests/fixtures/r_no_lockfile/code/analysis.R
library(ggplot2)
library(dplyr)
library(haven)

# Load data — absolute path (bad practice, left over from author's machine)
df <- read_dta("/Users/jsmith/Dropbox/JPE_paper/data/lfs_2019.dta")

df_clean <- df %>% filter(!is.na(wage))

p <- ggplot(df_clean, aes(x = age, y = wage)) +
  geom_point() +
  theme_minimal()

# Writes to working directory, not the figures/ folder LaTeX expects
ggsave("fig1.pdf", plot = p, width = 8, height = 5)

File: tests/fixtures/python_project/code/analysis.py
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
import numpy as np

# Load data — absolute path (bad practice, left over from author's machine)
df = pd.read_csv("/Users/jsmith/Dropbox/JPE_paper/data/lfs_2019.csv")

df = df[df["wage"].notna()].copy()

# Fit model
X = df[["age"]].values
y = df["wage"].values
model = LinearRegression().fit(X, y)

# Plot
fig, ax = plt.subplots()
ax.scatter(df["age"], df["wage"], alpha=0.3)
ax.plot(df["age"], model.predict(X), color="red")
ax.set_xlabel("Age")
ax.set_ylabel("Wage")

# Writes to working directory, not the figures/ folder LaTeX expects
plt.savefig("fig1.pdf")
